tests/gold_tests/pluginTest/header_rewrite/header_rewrite_set_body_from.test.py

```diff
@@ -0,0 +1,160 @@
+        remap_fail_1_request_header = {"headers": "GET /remap_fail HTTP/1.1\r\nHost: www.example.com\r\n\r\n"}
+        self.server.addResponse("sessionfile.log", remap_fail_1_request_header, response_header)
+
+        plugin_fail_1_request_header = {"headers": "GET /plugin_fail HTTP/1.1\r\nHost: www.example.com\r\n\r\n"}
+        self.server.addResponse("sessionfile.log", plugin_fail_1_request_header, response_header)
+
+        # Request/response for original successful transaction with successful second tranasaction
+        remap_success_1_request_header = {"headers": "GET /remap_success HTTP/1.1\r\nHost: www.example.com\r\n\r\n"}
+        self.server.addResponse("sessionfile.log", remap_success_1_request_header, response_header)
+
+        plugin_success_1_request_header = {"headers": "GET /plugin_success HTTP/1.1\r\nHost: www.example.com\r\n\r\n"}
+        self.server.addResponse("sessionfile.log", plugin_success_1_request_header, response_header)
+
+        # Request/response for custom body transaction that fails to retrieve body
+        fail_2_request_header = {"headers": "GET /502 HTTP/1.1\r\nHost: www.example.com\r\n\r\n"}
+        fail_2_response_header = {"headers": "HTTP/1.1 502 \r\nConnection: close\r\n\r\n", "body": "Fail\n"}
+        self.server.addResponse("sessionfile.log", fail_2_request_header, fail_2_response_header)
+
+        # Request/response for custom body transaction that successfully retrieves body
+        success_2_request_header = {"headers": "GET /404.html HTTP/1.1\r\nHost: www.example.com\r\n\r\n"}
+        success_2_response_header = {"headers": "HTTP/1.1 200 OK\r\nConnection: close\r\n\r\n", "body": "Custom body found\n"}
+        self.server.addResponse("sessionfile.log", success_2_request_header, success_2_response_header)
+
+    def setUpTS(self):
+        self.ts = Test.MakeATSProcess("ts")
+
+        # Set header rewrite rules
+        self.ts.Setup.CopyAs('rules/rule_set_body_from_remap.conf', Test.RunDirectory)
+        self.ts.Setup.CopyAs('rules/rule_set_body_from_plugin.conf', Test.RunDirectory)
+        self.ts.Disk.remap_config.AddLine(
+            """\
+             map http://www.example.com/remap_success http://127.0.0.1:{0}/remap_success @plugin=header_rewrite.so @pparam={1}/rule_set_body_from_remap.conf
+             map http://www.example.com/200 http://127.0.0.1:{0}/200 @plugin=header_rewrite.so @pparam={1}/rule_set_body_from_remap.conf
+             map http://www.example.com/remap_fail http://127.0.0.1:{0}/remap_fail @plugin=header_rewrite.so @pparam={1}/rule_set_body_from_remap.conf
+             map http://www.example.com/plugin_success http://127.0.0.1:{0}/plugin_success
+             map http://www.example.com/plugin_fail http://127.0.0.1:{0}/plugin_fail
+             map http://www.example.com/404.html http://127.0.0.1:{0}/404.html
+             map http://www.example.com/502 http://127.0.0.1:{0}/502
+             """.format(self.server.Variables.Port, Test.RunDirectory))
```

Comment:
I suggest you also have a remap for a body URL like this:
```

Test.GetTcpPort("bad_port")
self.ts.Disk.remap_config.AddLine(f'map http://www.example.com/plugin_no_server http://127.0.0.1::{Test.Variables.bad_port}/plugin_no_server")
```
in order to verify proper behavior if it's not possible to connect to the server that provided the body that's set by the operator.